packages/property-provider/src/memoize.spec.ts

```diff
@@ -20,22 +20,28 @@ describe("memoize", () => {
       const memoized = memoize(provider);
       expect(provider).toHaveBeenCalledTimes(0);
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      for (const index in [...Array(repeatTimes).keys()]) {
+      for (const _ in [...Array(repeatTimes).keys()]) {
         expect(await memoized()).toStrictEqual(mockReturn);
         expect(provider).toHaveBeenCalledTimes(1);
       }
     });
 
-    it("should always return the same promise", () => {
```

Comment:
This unit test is removed. We cannot return the same promise because the memoize() now needs to await the provider to resolve(). The unit test was introduced in https://github.com/aws/aws-sdk-js-v3/pull/56 but with no explicit purpose. I have run the integration test to make sure it feature doesn't affect the SDK now.